app/services/holdings_request_service.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+class HoldingsRequestService
+  attr_reader :args, :locations, :libraries, :exclude
+
+  def initialize(args)
+    @args = args
+    libraries_data = JSON.parse(File.read(Rails.root.join('app/javascript/availability/libraries_locations.json')))
```

Comment:
@Smullz622 a small refactor recommendation here maybe to move this to a one line method sth like `libraries_data`

```suggestion
  @libraries_data = libraries_data
```

```
    def libraries_data
      @libraries_data ||= JSON.parse(File.read(Rails.root.join('app/javascript/availability/libraries_locations.json')))
    end
```

Reply:
Done! Since it's only used to initialize other variables, I pared it down a little bit.